gmso/formats/networkx.py

```diff
@@ -0,0 +1,953 @@
+            print("Dihedral not typed")
+    elif len(param) == 3:
+        try:
+            for angle in list(topology.angles):
+                if angle.connection_members == (param[0], param[1], param[2]):
+                    print(angle.angle_type.expression, "\n")
+                    print("{:<12} {:<15}".format("Parameter", "Value"))
+                    for k, v in angle.angle_type.parameters.items():
+                        print("{:<12} {:<15}".format(k, v))
+        except AttributeError:
+            print("Angle not typed")
+    else:
+        raise ValueError(
+            "Parameters are not proper angles or dihedrals. Connection members are missing"
+        )
+
+    return
+
+
+def get_edges(networkx_graph, atom_name1, atom_name2):
+
+    # Create a list of the edges that have been selected corresponding to the selected atom_name1
+    # and atom_name2. If both are None, then select edges that are missing bond types.
+    list_of_edges = []
+    list_of_bonds = []
+    # Check for whether you want to plot bonds based on the atom types, or their node labels.
+    try:
+        [node.atom_type.name for node in networkx_graph.nodes]
+    except AttributeError:
+        return print("Atomtypes are missing, so no bond types can be selected")
+    mia_bond_flag = 0
+    if atom_name1 and atom_name2:
+        for edge in list(networkx_graph.edges):
+            if (
+                edge[0].name == atom_name1
+                and edge[1].name == atom_name2
+                or edge[0].name == atom_name2
+                and edge[1].name == atom_name1
+            ):
+                list_of_bonds.append(
+                    edge[0].atom_type.name + " --- " + edge[1].atom_type.name
+                )
+                list_of_edges.append(edge)
+    elif atom_name1:
+        for edge in list(networkx_graph.edges):
+            if edge[0].name == atom_name1 or edge[1].name == atom_name1:
+                list_of_bonds.append(
+                    edge[0].atom_type.name + " --- " + edge[1].atom_type.name
+                )
+                list_of_edges.append(edge)
+    else:
+        for nodes in list(networkx_graph.edges.items()):
+            if nodes[1]["connection"].bond_type is None:
+                list_of_bonds.append(
+                    nodes[0][0].atom_type.name + " --- " + nodes[0][1].atom_type.name
+                )
+                list_of_edges.append((nodes[0][0], nodes[0][1]))
+                mia_bond_flag = 1
+        if not mia_bond_flag:
+            return print("All bonds are typed")
+
+    labeled_bonds = zip(list_of_bonds, list_of_edges)
+
+    # create a dic so each selected bond selects the proper edges on the networkx.graph object.
+    selectable_list = {}
+    for label, bond in labeled_bonds:
+        if label in selectable_list.keys():
+            selectable_list[label].append(bond)
+        else:
+            selectable_list[label] = []
+            selectable_list[label].append(bond)
```

Comment:
Why not build this dictionary in the above loops?

Reply:
This more or less just takes the zipped lists of edges and the corresponding name of that edge, and just thins it out so repeats are not shown. Otherwise, each bond that had the same label would be shown in the dropdown selection, and would clutter it up with repeats. So the above loops generate the full list, and the second loop shown here goes through and finds only unique examples to put in the dictionary. I could simplify this using `numpy.unique()` I think.